### PR TITLE
[wg21-linear-algebra] Add new port

### LINKED ISSUE #28379: [New Port Request] linear_algebra
Library name: linear_algebra

Library description: A linear algebra proposal for the C++ standard library

Source repository URL: https://github.com/BobSteagall/wg21

### PR DESCRIPTION
**Pull request by globberwops**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #28379

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by globberwops** [comment]
@microsoft-github-policy-service agree

File: ports/wg21-linear-algebra/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO BobSteagall/wg21
+    REF v0.7.2
+    SHA512 c249344d035d09760a9e5ea059ed6db5a1cb42b918735672bd7aa6dbda08f947855582f76ad61d33f59a847d8befe5caed57d25da2bcfc9fa8e6cef50a4c24e2
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH "${SOURCE_PATH}"
+    PREFER_NINJA
+    OPTIONS -DLA_INSTALL=1 -DLA_BUILD_PACKAGE=0 -DLA_ENABLE_TESTS=0
```
Comment (LilyWangLL) [inline ports/wg21-linear-algebra/portfile.cmake:11]:
```suggestion
vcpkg_cmake_configure(
    SOURCE_PATH "${SOURCE_PATH}"
    OPTIONS
        -DLA_INSTALL=ON
        -DLA_BUILD_PACKAGE=OFF
        -DLA_ENABLE_TESTS=OFF
```

File: ports/wg21-linear-algebra/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+vcpkg_install_cmake()
```
Comment (LilyWangLL) [inline ports/wg21-linear-algebra/portfile.cmake:14]:
```suggestion
vcpkg_cmake_install()
```

File: ports/wg21-linear-algebra/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,23 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO BobSteagall/wg21
+    REF v0.7.2
```
Comment (LilyWangLL) [inline ports/wg21-linear-algebra/portfile.cmake:4]:
```suggestion
vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}

vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO BobSteagall/wg21
    REF "v${VERSION}"
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by globberwops** [comment]
@LilyWangLL Thank you for you feedback. Fixed!

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**